echopype/calibrate/api.py

```diff
@@ -34,11 +58,159 @@ def _compute_cal(
+    EK60, AZFP, EK80 (see Notes below for detail).
+
+    Parameters
+    ----------
+    echodata : EchoData
+        An `EchoData` object created by using `open_raw` or `open_converted`
+
+    env_params : dict
+        Environmental parameters needed for calibration.
+        Users can supply `"sound speed"` and `"absorption"` directly,
+        or specify other variables that can be used to compute them,
+        including `"temperature"`, `"salinity"`, and `"pressure"`.
+
+        For EK60 and EK80 echosounders, by default echopype uses
+        environmental variables stored in the data files.
+        For AZFP echosounder, all environmental parameters need to be supplied.
+        AZFP echosounders typically are equipped with an internal temperature
+        sensor, and some are equipped with a pressure sensor, but automatically
+        using these pressure data is not currently supported.
+
+    cal_params : dict
+        Intrument-dependent calibration parameters.
+
+        - for EK60 echosounder, allowed parameters include:
+          `"sa_correction"`, `"gain_correction"`, `"equivalent_beam_angle"`
+        - for AZFP echosounder, allowed parameters include:
+          `"EL"`, `"DS"`, `"TVR"`, `"VTX"`, `"equivalent_beam_angle"`, `"Sv_offset"`
+
+        Passing in calibration parameters for other echosounders
+        are not currently supported
+
+    waveform_mode : {"CW", "BB"}, optional
+        Type of transmit waveform.
+        Required only for data from the EK80 echosounder.
+
+        - `"CW"` for narrowband transmission,
+          returned echoes recorded either as complex or power/angle samples
+        - `"BB"` for broadband transmission,
+          returned echoes recorded as complex samples
+
+    encode_mode : {"complex", "power"}, optional
+        Type of encoded return echo data.
+        Required only for data from the EK80 echosounder.
+
+        - `"complex"` for complex samples
+        - `"power"` for power/angle samples, only allowed when
+          the echosounder is configured for narrowband transmission
+
+    Returns
+    -------
+    xr.Dataset
+        The calibrated Sp dataset, including calibration parameters
+        and environmental variables used in the calibration operations.
+
+    Notes
+    -----
+    The EK80 echosounder can be configured to transmit
+    either broadband (``waveform_mode="BB"``)
+    or narrowband (``waveform_mode="CW"``) signals.
+    When transmitting in broadband mode, the returned echoes are
+    encoded as complex samples (``encode_mode="complex"``).
+    When transmitting in narrowband mode, the returned echoes can be encoded
+    either as complex samples (``encode_mode="complex"``)
+    or as power/angle combinations (``encode_mode="power"``) in a format
+    similar to those recorded by EK60 echosounders.
+
+    The current calibration implemented for EK80 broadband complex data
+    uses band-integrated Sv with the gain computed at the center frequency
```

Comment:
Should that be Sp rather than Sv? (I don't know enough about Sp to know)

Reply:
Yes, changed.